data/blog/2025/2025-01-30-mezzio101-using-mezzio-skeleton-installer.md

```diff
@@ -0,0 +1,173 @@
+
+### What is Mezzio?
+
+As mentioned on [the Zend official site](https://www.zend.com/resources/what-mezzio-why-would-i-use-it), 'Mezzio is the Laminas middleware runtime -- previously known as Expressive in the Zend framework.'
+Mezzio is designed around PHP-FIG, the PHP Framework Interop Group, that has defined several standards recommendations.
+The goal of the standards is to ensure that PHP code is written in a consistent, interoperable, and maintainable way.
+
+### What is middleware?
+
+Middleware is code that is executed between the request and response.
+Most commonly, it performs these tasks, but more layers can be defined based on your application:
+
+- Aggregates incoming data.
+- Processes the request.
+- Creates and returns a response.
+
+There are several advantages to using middleware:
+
+- It enables old systems to interface with newer ones.
+- It's scalable.
+- Integrations are created more easily.
+- It enables automating processes.
+
+### Installing Mezzio
+
+Let's put aside the technical stuff in favor of some real work that you can be proud of.
+Building things with our hands, kind of.
+
+To start, create a folder where you want your project files to be created.
+
+```shell
+mkdir -p ~/Projects/my-mezzio-project
+```
+
+The next command will launch the Mezzio Skeleton Installer.
+We will go through the installation process together, using the recommended settings.
+We will opt primarily for the Laminas-supported options to maintain consistency.
+Funnily enough, the installation should complete in less time than it takes for you to read this tutorial.
+
+```shell
+composer create-project mezzio/mezzio-skeleton ~/Projects/my-mezzio-project
+```
+
+You should see the text and prompt below:
+
+![step-1-installation-type](/images/blog/mezzio101/mezzio101-use-skeleton-01.jpg "step-1")
+
+Submit `3` for option `Modular`.
+
+![step-2-dependency-injection](/images/blog/mezzio101/mezzio101-use-skeleton-02.jpg "step-2")
+
+Submit `1` for option `laminas-servicemanager`.
+Just as a side note, this option is Laminas's factory-driven dependency injection container.
+Its version 4.0 supports PSR-11: Container interface version 1.1 and 2.
+
+![step-3-router](/images/blog/mezzio101/mezzio101-use-skeleton-03.jpg "step-3")
+
+Submit `1` for `fastroute`.
+
+![step-4-templating](/images/blog/mezzio101/mezzio101-use-skeleton-04.jpg "step-4")
+
+We chose `Twig` as the templating engine because of its concise syntax, extensibility and well-written documentation.
+The Mezzio Skeleton Installer does offer several alternatives for the templating engines:
+
+- `Plates` is a simple templating engine that sees the templates themselves as PHP files.
+- `Twig` is reliable, actively-developed and well-known in the PHP community.
+- `laminas-view` is related to Laminas MVC.
+
+The default option for this step doesn't install any templating engine, but we only recommend this option if you want to build a CLI or API.
```

Comment:
You might use Laminas components for CLI applications, but Mezzio, not so much :)

Reply:
I mean, why would you use the Mezzio skeleton to scaffold out a CLI app? The CLI bit needs removing so it's just:

> The default option for this step doesn't install any templating engine which is only relevant if you are building an API.